pkg/virtual/initializingworkspaces/builder/caching_authorizer.go

```diff
@@ -0,0 +1,71 @@
+package builder
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	kcpkubernetesclientset "github.com/kcp-dev/client-go/kubernetes"
+	"github.com/kcp-dev/kcp/pkg/apis/tenancy/initialization"
+	tenancyv1alpha1 "github.com/kcp-dev/kcp/pkg/apis/tenancy/v1alpha1"
+	"github.com/kcp-dev/kcp/pkg/authorization/delegated"
+	dynamiccontext "github.com/kcp-dev/kcp/pkg/virtual/framework/dynamic/context"
+	"github.com/kcp-dev/logicalcluster/v2"
+	"k8s.io/apiserver/pkg/authorization/authorizer"
+	"k8s.io/klog/v2"
+)
+
+func newCachingAuthorizer(client kcpkubernetesclientset.ClusterInterface) authorizer.Authorizer {
+	return &cachingAuthorizer{client: client, cache: make(map[string]authorizer.Authorizer)}
+}
+
+type cachingAuthorizer struct {
+	client kcpkubernetesclientset.ClusterInterface
+
+	cacheMu sync.RWMutex
+	cache   map[string]authorizer.Authorizer
+}
+
+func (c *cachingAuthorizer) loadOrStore(workspace logicalcluster.Name) (authorizer.Authorizer, error) {
+	c.cacheMu.RLock()
+	if authz, ok := c.cache[workspace.String()]; ok && authz != nil {
```

Comment:
Is a key by workspace enough in this case?

Reply:
With how you have it implemented, I'd say 👍